design-document.md

```diff
@@ -106,6 +123,167 @@ to achieve these goals:
+   clients or coalitions of clients from compromising the robustness property.
+
+#### Aggregator
+
+##### Assets
+
+1. Unencrypted data shares.
+1. Data share decryption keys.
+1. Client identifying information.
+1. Aggregation parts.
+1. Aggregator identity.
+
+##### Capabilities
+
+1. Aggregators may defeat the robustness of the system by emitting bogus
+   aggregation shares.
+1. If clients reveal identifying information to aggregators (such as a trusted
+   identity during client authentication), aggregators can learn which clients
+   are contributing data.
+     1. Aggregators may weaken anonymity by revealing that a particular client
+        contributed data to the system.
+     1. Aggregators may choose to selectively omit data from certain clients.
+          * For example, omitting submissions from a particular geographic
+            region to falsely suggest that a particular localization is not
+            being used.
+1. Individual aggregators may compromise availability of the system by refusing
+   to emit aggregation parts.
+
+##### Mitigations
+
+1. The linear secret sharing scheme employed by the client ensures that privacy
+   is preserved unless every participating aggregator reveals their unencrypted
+   data shares.
+1. Infrastructure diversity. Prio deployments should ensure that aggregators do
+   not have common dependencies that would enable a single vendor to reassemble
+   data.
+     * For example, if all participating aggregators stored unencrypted data
+       shares on the same cloud object storage service, then that cloud vendor
+       would be able to reassemble all the data shares and defeat privacy.
+1. Running the protocol over multiple subsets of the available aggregators
+   chosen so that no aggregator appears in all subsets and accepting any result
+   would prevent an individual aggregator from compromising availability.
+1. Running the protocol over multiple subsets of the available aggregators
+   chosen so that no aggregator appears in all subsets and requiring that the
+   results agree would prevent an individual aggregator from compromising
+   robustness
+     * This could also allow identifying of defective aggregators, by finding
+       the aggregator that appears in every subset whose results were wrong.
+
+#### Leader
+
+The leader is also an aggregator, and so all the assets, capabilities and
+mitigations available to aggregators also apply to the leader.
+
+##### Capabilities
+
+1. Data validity proof verification. The leader can forge proofs and collude
+   with a malicious client to trick aggregators into aggregating invalid data.
+1. Relaying messages between aggregators. The leader can compromise availability
+   by dropping messages.
+
+##### Mitigations
+
+1. Rather than having all aggregators trust the leader to validate proofs, each
+   aggregator could exchange proof shares with every other aggregator, enabling
+   each aggregator to independently verify the proofs.
```

Comment:
I think what's meant by this is that the aggregators could exchange verification shares so that each aggregator can run the decision algorithm locally. Is that right?

I agree that this would help mitigate the leader's additional capabilities. However, doing so would require a secure channel between each pair of aggregators. I think this cost is too high.

Reply:
HCG convinced us (or at least me!) that it's OK to have a designated proof verifier because the verifier is no more powerful than any aggregator that colludes with a malicious client, so I'm going to remove this item.